drivers/sensor/mcp9600/mcp9600.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2020 SER Consulting, LLC
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * Note, adapted from many drivers, Trigger functionality not tested yet.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <kernel.h>
+#include <drivers/i2c.h>
+#include <init.h>
+#include <sys/byteorder.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include "mcp9600.h"
+
+#define DT_DRV_COMPAT microchip_mcp9600
+
+#define MCP9600_BUS_I2C DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+
+LOG_MODULE_REGISTER(MCP9600, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "MCP9600 driver enabled without any devices"
+#endif
+
+static int mcp9600_reg_read(const struct device *dev, uint8_t start,
+			uint8_t *buf, int size)
+{
+	struct mcp9600_data *data = dev->data;
+	const struct mcp9600_config *cfg = dev->config;
+
+	return i2c_burst_read(data->i2c_master, cfg->i2c_addr,
+						start, buf, size);
+}
+
+static int mcp9600_sample_fetch(const struct device *dev,
+			enum sensor_channel chan)
+{
+	struct mcp9600_data *data = dev->data;
+	uint8_t buf[8];
+	int size = 6;
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	ret = mcp9600_reg_read(dev, MCP9600_REG_TEMP_HOT, buf, size);
+	if (ret < 0) {
+		return ret;
+	}
+	data->ttemp = (buf[0]<<8) | (buf[1]);
```

Comment:
I believe spaces around the `<<` operator is more standard, though that might not be present in all existing code.  The array reference operand `buf[1]` doesn't need parentheses around it; the precedence should be clear enough for anybody even if they don't remember whether `<<` or `|` is higher.

Reply:
ok, check patch seems not to care, but I can do that.